src/cephadm/cephadm.py

```diff
@@ -6286,7 +6286,7 @@ def _parse_ipv4_route(out: str) -> Dict[str, Dict[str, Set[str]]]:
         if '/' not in net:  # aggregate /32 mask for single host sub-networks
             net += '/32'
         iface = m[0][1]
-        ip = m[0][4]
+        ip = get_ipv4_address(iface)
```

Comment:
there's a handful of unit tests we have around network testing that seem to be failing with this change. If you go to `ceph/src/cephadm` and run `tox` on a local branch with your changes it will run the tests and you can try getting the change to work with the tests (or changing the tests if you think they're incorrect). From what I can see
`test_parse_ipv4_route` and `test_command_list_networks` in `ceph/src/cephadm/tests/test_networks.py` are failing.

Reply:
tests are written in such a way that it always tries to get the value of an IP using the the list of tuple i.e m[0][4]. But when vlan interfaces are used, we cannot rely on the list of tuple as the tuple will not have the IP address in it. 

ex:
`10.10.35.0/25 dev eno6.20 scope link
10.180.21.128/26 dev eno5 proto kernel scope link src 10.180.21.141
10.180.37.32/27 dev eno5.10 scope link
172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 linkdown`


as per the code, - the value of "m" will be - 
`[('10.180.21.128/26', 'eno5', 'proto kernel ', '', '10.180.21.141')]
[('172.17.0.0/16', 'docker0', 'proto kernel ', '', '172.17.0.1')]`

and this line - https://github.com/ceph/ceph/blob/main/src/cephadm/cephadm.py#L6289  will fail to get the IP address for eno5.10 and eno6.20 interface as value of "m" will not have IP address in it. 

Considering this, the value of "ip" variable can be assigned directly using **"get_ipv4_address"**  via socket library.  


The tests always fail as it tries to get the value of variable "ip" using the test_input which is static. But as per this PR we are using get_ipv4_address function to assign correct value to the "ip". This is dynamic in nature based on the host we run.

Reply:
okay, well if the test is not compatible with the new way of doing things, the test will need to be updated as part of this change.